oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexWriter.java

```diff
@@ -97,29 +98,30 @@ public boolean close(long timestamp) {
         return bulkProcessorHandler.close();
     }
 
-    protected void provisionIndex() throws IOException {
+    protected void provisionIndex(long seed) throws IOException {
         // check if index already exists
+        final String indexName = ElasticIndexNameHelper.getElasticSafeIndexName(
```

Comment:
IMHO index naming logic should be centralized in one place. The logic is spread across `ElasticIndexHelper`, `ElasticIndexDefinition` and `ElasticIndexEditorContext`.

What about this:
* ElasticIndexDefinition exposes a method `createUniqueIndexName` (it generates the seed, saves the seed and concatenate it to the alias returning the complete name)
* ElasticIndexEditorContext invokes the above method in `enableReindexMode` and calls `provisionIndex` with the complete name instead of the seed
* ElasticIndexEditorContext does not need to compose the index name since it receives the complete one in the method signature

Reply:
The problem is that even if we generate seed in `ElasticIndexDefinition`, we can't save it to repository there because it doesn't have any `NodeBuilder` instance. `ElasticIndexDefinition` only has `NodeState`s which are immutable. So I thought that an index definition object is just designed to read but not write anything to repository and any write operation should go in editor context (I see many write operations in `FullTextIndexEditorContext`)